tiledb/sm/subarray/range_subset.h

```diff
@@ -141,7 +140,78 @@ struct SortStrategy<std::string, std::string> {
           return a.start_str() < b.start_str() ||
                  (a.start_str() == b.start_str() && a.end_str() < b.end_str());
         });
-    return Status::Ok();
+  };
+};
+
+/** Default merge behavior: merging is not enabled. */
+template <typename T, typename Enable = T>
+struct MergeStrategy {
+  static void merge(std::vector<Range>&);
+};
+
+/** Merge algorithm for numeric-type ranges. */
+template <typename T>
+struct MergeStrategy<
+    T,
+    typename std::enable_if<std::is_arithmetic<T>::value, T>::type> {
+  static void merge(std::vector<Range>& ranges) {
+    auto head = ranges.begin();
+    size_t merged_cells = 0;
+
+    // Merge
+    for (auto tail = head + 1; tail != ranges.end(); tail++) {
+      const bool can_coalesce = *static_cast<const T*>(head->end_fixed()) !=
+                                    std::numeric_limits<T>::max() &&
+                                *static_cast<const T*>(head->end_fixed()) + 1 ==
+                                    *static_cast<const T*>(tail->start_fixed());
+      const bool can_merge = *static_cast<const T*>(head->start_fixed()) <=
+                                 *static_cast<const T*>(tail->start_fixed()) &&
```

Comment:
`*static_cast<const T*>(head->start_fixed()) <= *static_cast<const T*>(tail->start_fixed()`

I think this will always be true if the ranges are sorted? We then might be able to merge the two conditions into:
(head->end() != MAX) && (tail->start() <= head->end() + 1)

Reply:
Note that float and doubles cannot be coalesced... We might need a different merge strategy or we could make the +1 in the condition I added above only present in integer conditions with std::is_integral<T>.